registry.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package credentials
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"oras.land/oras-go/v2/registry/remote"
+	"oras.land/oras-go/v2/registry/remote/auth"
+)
+
+// Login provides the login functionality with the given credentials. A local client is used
+// to keep the original client of the registry intact.
```

Comment:
"A local client is used to keep the original client of the registry intact." requires rewording.

BTW, we require the `reg.Client` to be nil or of `*auth.Client` type. Basically, developers will not read your code but read your documentation before invoking the API.

Reply:
changed the comment.